### PR TITLE
feat: Add base data controller

### PR DESCRIPTION
**Pull request by oskarwilliams378**

### CODE REVIEW

File: src/main/kotlin/uk/gov/gdx/datashare/controller/BaseDataController.kt [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    summary = "Get Ingress Events",
+    description = "Need scope of pubsub/maintain",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Ingress Event"
+      )
+    ]
+  )
+  suspend fun getIngressEvents() = ingressEventDataRepository.findAll().toList()
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/gdx/datashare/controller/BaseDataController.kt:158]: Do you need the toList?  Return a flow instead?

**Review by Mjwillis — COMMENTED** [review]
Flow comment

File: src/main/kotlin/uk/gov/gdx/datashare/controller/BaseDataController.kt (hunk 1 above)
Comment (oskarwilliams378) [reply src/main/kotlin/uk/gov/gdx/datashare/controller/BaseDataController.kt:158]: removed the to list :)

File: src/main/kotlin/uk/gov/gdx/datashare/controller/AdminController.kt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package uk.gov.gdx.datashare.resource
```
Comment (olevett) [inline src/main/kotlin/uk/gov/gdx/datashare/controller/AdminController.kt:1]: I think I'd ideally have done this rename as a completely separate commit to main - it makes this a little more noisy than would be ideal

File: src/main/kotlin/uk/gov/gdx/datashare/controller/BaseDataController.kt [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+      )
+    ]
+  )
+  suspend fun deletePublisherSubscription(
+    @Schema(description = "Publisher Subscription ID", required = true)
+    @PathVariable id: UUID,
+  ) = publisherSubscriptionRepository.deleteById(id)
+
+  //The below endpoints don't exist through any other means, so have added them here
+  @PostMapping("/ingressType")
```
Comment (olevett) [inline src/main/kotlin/uk/gov/gdx/datashare/controller/BaseDataController.kt:264]: I think we need to make sure we've fully aligned around "ingress", "publisher" etc - but that's out of this scope I think

**Review by olevett — APPROVED** [review]

File: src/main/kotlin/uk/gov/gdx/datashare/controller/AdminController.kt (hunk 2 above)
Comment (oskarwilliams378) [reply src/main/kotlin/uk/gov/gdx/datashare/controller/AdminController.kt:1]: splitting

File: src/main/kotlin/uk/gov/gdx/datashare/controller/BaseDataController.kt (hunk 3 above)
Comment (oskarwilliams378) [reply src/main/kotlin/uk/gov/gdx/datashare/controller/BaseDataController.kt:264]: I totally agree, this is just to keep the mapping consistent atm :)
